packages/webdriverio/src/commands/element/isDisplayed.js

```diff
@@ -34,20 +34,38 @@
  * </example>
  *
  * @alias element.isDisplayed
- * @return {Boolean} true if element(s)* [is|are] displayed
+ * @return {Boolean} true if element is displayed
  * @uses protocol/elements, protocol/elementIdDisplayed
  * @type state
  *
  */
 
+import { ELEMENT_KEY } from '../../constants'
+import { getBrowserObject } from '../../utils'
+import isElementDisplayedScript from '../../scripts/isElementDisplayed'
+
 export default async function isDisplayed() {
 
+    let browser = getBrowserObject(this)
+
     /*
      * This is only necessary as isDisplayed is on the exclusion list for the middleware
      */
     if (!this.elementId) {
         this.elementId = (await this.parent.$(this.selector)).elementId
     }
 
-    return this.elementId ? await this.isElementDisplayed(this.elementId) : false
+    /*
+     * if element was still not found it also is not displayed
+    */
+    if (!this.elementId) {
+        return false
+    }
+
+    return browser.isChrome ?
```

Comment:
**This change it going to break all native app testing.** 
There should be a way to determine this is or Safari / Edge. Secondly we need to be aware of the fact that SafariDriver is also used for Hybrid apps. 

Challenges we have with apps:
We need to determine the if it is an app, but then we also got a challenge because of the following reasons

- app can be native, so no webview, and not using safari / chrome driver
- app is provided by the `app`-capability by default but there are cloud vendors that provide an extra capability to provide the app to there cloud, so we need to check multiple capabilities to determine if is is an app.

So determining if this is an app will be challenging. We might need to contact the Appium team to see if there is an easy way to determine if the safari driver is used during automation. Because you can also switch between context (native, where the XCUITest driver is user, or Webdriver where the safari driver is used.) during your tests

Reply:
Nice catch!

Reply:
Thanks @wswebcreation ! I wasn't exactly happy with this solution anyways, as I assumed it would cause more headaches than benefit.

Reply:
@wswebcreation Please re-review/ retest. 

I updated the logic to only run on Safari and Edge when running in w3c mode.  I'm honestly not really sure what to do about hybrid apps:
1.) If they are using safaridriver then the isDisplayed api no longer exists
2.) If they don't support execute, then they are kind of SOL

I was testing this last night, and I think we need to do a better job of determining who is w3c though. This Fix works on Edge 18 and Safari 12, but doesn't work on the previous versions of either (and I don't have the cleanest way to sniff the version other than browser.isW3C).